src/main/java/couchbase/test/val/Product.java

```diff
@@ -0,0 +1,124 @@
+
+    public void setReviewsArray() {
+        int numReviews = this.random.nextInt(10);
+        LocalDateTime now = LocalDateTime.now();
+        ArrayList<JsonObject> temp = new ArrayList<JsonObject>();
+        for (int n = 0; n <= numReviews; n++) {
+            JsonObject review = JsonObject.create();
+            review.put("author", faker.name().fullName());
+            review.put("date", now.plus(n, ChronoUnit.WEEKS).format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")));
+            JsonObject ratings = JsonObject.create();
+            ratings.put("rating_value", this.random.nextInt(10));
+            ratings.put("performance", this.random.nextInt(10));
+            ratings.put("utility", this.random.nextInt(10));
+            ratings.put("pricing", this.random.nextInt(10));
+            ratings.put("build_quality", this.random.nextInt(10));
+            review.put("ratings", ratings);
+            temp.add(review);
+        }
+        this.reviews.add(temp);
+    }
+    
+    public Product(WorkLoadSettings ws) {
+        super();
+        this.ws = ws;
+        this.random = new Random();
+        this.random.setSeed(ws.keyPrefix.hashCode());
+        faker = new Faker(this.random);
+        for (int index=0; index<4096; index++) {
+            this.setReviewsArray();
+        }
+    }
+
+    public  ArrayList<String> buildProductCategory(int length) {
+    	ArrayList<String> prodCategory = new ArrayList<String>();
+    	for (int n = 0; n <= length; n++) {
+    		prodCategory.add(faker.book().genre());
+    	}
+    	return prodCategory;
+    }
+
+    public ArrayList<String> buildProductImageLinks(int length) {
+    	ArrayList<String> prodImgLinks = new ArrayList<String>();
+    	for (int n = 0; n <= length; n++) {
+    		prodImgLinks.add(faker.internet().url());
+    	}
+    	return prodImgLinks;
+    }
+
+    public ArrayList<String> buildProductFeatures(int length) {
+    	ArrayList<String> prodFeatures = new ArrayList<String>();
+    	for (int n = 0; n <= length; n++) {
+    		prodFeatures.add(faker.book().genre());
+    	}
+    	return prodFeatures;
+    }
+
+    public JsonObject buildProductSpecs() {
+    	JsonObject prodSpecs = JsonObject.create();
+    	
+    	prodSpecs.put("make", faker.book().author());
+    	prodSpecs.put("model", faker.book().title());
+
+    	return prodSpecs;
+    }
+    
+    public JsonObject next(String key) {
+        this.random = new Random();
+        JsonObject jsonObject = JsonObject.create();
+        this.random.setSeed((key).hashCode());
+        if(this.ws.mutated > 0){
+            this.random.setSeed((key+Integer.toString(this.ws.mutated)).hashCode());
+        }
+        int index = this.random.nextInt(4096);
+        jsonObject.put("id", "P" + key);
+        jsonObject.put("product_name", faker.name().firstName());
+        jsonObject.put("product_link", faker.internet().url());
+        jsonObject.put("product_features", this.buildProductFeatures(1 + this.random.nextInt(3)));
```

Comment:
Looks good to me, but a couple questions - 

seller_verified is something you've added? I don't see it in the existing template.

Should we also have some docs with 0 length arrays for product links/ reviews etc? Just for the sake of variety.

Reply:
I was looking at an older version of the template. seller_verified is part of the new one. I've struck it from my  comment.

Reply:
Product link is a string not an array and for reviews, products with 0 reviews are already considered.